BackEnd/src/main/java/com/exadel/carpoolfree/controller/DriveController.java

```diff
@@ -15,29 +18,63 @@
 import org.springframework.web.bind.annotation.RestController;
 
 import java.time.LocalDateTime;
+import java.util.LinkedList;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
 @RestController
 @RequestMapping("api/drive")
 public class DriveController {
     private final DriveRepository driveRepository;
+    private final PassengerDriveRepository passengerDriveRepository;
     private final PathRepository pathRepository;
 
-    public DriveController(DriveRepository driveRepository, PathRepository pathRepository) {
+    public DriveController(DriveRepository driveRepository, PassengerDriveRepository passengerDriveRepository, PathRepository pathRepository) {
         this.driveRepository = driveRepository;
+        this.passengerDriveRepository = passengerDriveRepository;
         this.pathRepository = pathRepository;
     }
 
     @GetMapping()
-    public List<Drive> findAllDrives() {
+    public List<Drive> findAllDrives()
+    {
         return driveRepository.findAll();
     }
 
+
     @GetMapping("/{id}")
     public Drive findById(final @PathVariable Long id) {
         return driveRepository.findById(id).get();
     }
 
+    @GetMapping("/driverId/{driverId}")
```

Comment:
this code belonds to service layer

Reply:
Where should service be? Should I create separate package with DriveService class in it?

Reply:
> Where should service be? Should I create separate package with DriveService class in it?

`service` package should be a package on the same level as `controler` or 'repository', just move the code there please